plugins/media/wowza/services/LiveConversionProfileService.php

```diff
@@ -174,16 +192,105 @@ public function serveAction($streamName, $hostname = null)
 		
 		return new kRendererString($dom->saveXML(), 'text/xml');
 	}
-
-	protected function appendLiveParams(LiveStreamEntry $entry, WowzaMediaServerNode $mediaServer = null, SimpleXMLElement $encodes, liveParams $liveParams)
+	
+	private function calculateFlavorHeight($flavorResolution, $ingestParameters)
+	{
+		return ($ingestParameters['width'] != 0) ? (($flavorResolution['width'] * $ingestParameters['height']) / $ingestParameters['width']) : 0;
+	}
+	
+	private function isFlavorCompatibile($ingestParameters, $flavorBitrate, $flavorResolution, $flavorId)
+	{
+		$flavorHeight = 0;
+		switch ($flavorResolution['fitMode'])
+		{
+			case 'match-source':
+				break;
+			case 'fit-height':
+				$flavorHeight = $flavorResolution['height'];
+				break;
+			case 'fit-width':
+				// Flavor's height is not defined in KMC, calculate it according to ingest/flavor ratio
+				$flavorHeight = $this->calculateFlavorHeight($flavorResolution, $ingestParameters);
+				break;
+		}
+		
+		if ($ingestParameters['height'] !== 0 && $ingestParameters['height'] < $flavorHeight)
+		{
+			$ingestResolutionString = $ingestParameters['width'] . 'x' . $ingestParameters['height'];
+			$flavorResolutionString = $flavorResolution['width'] . 'x' . $flavorHeight;
+			KalturaLog::info('Flavor [' . $flavorId . '] rejected due to Resolution; Ingest: [' . $ingestResolutionString . '], Flavor: [' . $flavorResolutionString . ']');
+			return false;
+		}
+		else if ($ingestParameters['videodatarate'] !== 0 && ($ingestParameters['videodatarate'] * 1024) < $flavorBitrate)
+		{
+			KalturaLog::info('Flavor [' . $flavorId . '] rejected due to VideoBitrate; Ingest: [' . $ingestParameters['videodatarate'] * 1024 . '], Flavor: [' . $flavorBitrate . ']');
+			return false;
+		}
+		
+		return true;
+	}
+	
+	private function getResolutionParameters($liveParams)
+	{
+		$resolutionObject = array(
+			'fitMode' => ''
+		);
+		if(!$liveParams->getWidth() && !$liveParams->getHeight())
```

Comment:
you can loose this check by simply defining the default value to 'match-source' instead of ''